data_layer/sqitch/deploy/plan_action/plan_action.sql

```diff
@@ -95,7 +223,544 @@ from fiche_action fa
+    'Fonction retournant un JSON contenant le plan d''action passé en paramètre,
+    et ses plans d''actions enfants de manière récursive';
+
+-- Vue pour afficher les axes d'un plan et leur profondeur
+create view plan_action_profondeur as
+select a.collectivite_id, a.id, plan_action_profondeur(a.id, 0) as plan
+from axe a
+where a.parent is null;
+
+-- Vue pour afficher le chemin d'un axe (plan - axe - sous-axe - etc.)
+create view plan_action_chemin as
+with recursive chemin_plan_action as (
+    select id as axe_id, collectivite_id,nom, parent, array[axe] as chemin
+    from axe
+    where parent is null
+
+    union all
+
+    select a.id as axe_id, a.collectivite_id, a.nom, a.parent, p.chemin || a
+    from axe a
+             join chemin_plan_action p on a.parent = p.axe_id
+)
+select chemin[1].id as plan_id, axe_id, collectivite_id, chemin
+from chemin_plan_action;
+
+-- Fonction récursive pour afficher un plan d'action, ses axes, et ses fiches
+create or replace function plan_action(id integer) returns jsonb as
+$$
+declare
+    pa_enfant_id integer; -- Id d'un plan d'action enfant du plan d'action courant
+    pa_axe axe; -- Axe courant
+    id_loop integer; -- Indice pour parcourir une boucle
+    enfants jsonb[]; -- Plans d'actions enfants du plan d'action courant;
+    fiches jsonb; -- Fiches actions du plan d'action courant
+    to_return jsonb; -- JSON retournant le plan d'action courant, ses fiches et ses enfants
+begin
+    fiches = to_jsonb((select array_agg(ff.*)
+                       from(
+                               select * from fiches_action fa
+                                                 join fiche_action_axe fapa on fa.id = fapa.fiche_id
+                               where fapa.axe_id = plan_action.id
+                               order by fa.modified_at desc
+                           )ff)) ;
+    select * from axe where axe.id = plan_action.id limit 1 into pa_axe;
+    id_loop = 1;
+    for pa_enfant_id in
+        select pa.id
+        from axe pa
+        where pa.parent = plan_action.id
+        order by pa.created_at asc
+        loop
+            enfants[id_loop] = plan_action(pa_enfant_id);
+            id_loop = id_loop + 1;
+        end loop;
+
+    to_return = jsonb_build_object('axe', pa_axe,
+                                   'fiches', fiches,
+                                   'enfants', enfants);
+    return to_return;
+end;
+$$ language plpgsql;
+comment on function plan_action is
+    'Fonction retournant un JSON contenant le plan d''action passé en paramètre,
+    ses fiches et ses plans d''actions enfants de manière récursive';
+
+-- Vue pour afficher les plan d'actions complet
+create view plan_action as
+select a.collectivite_id, a.id, plan_action(a.id) as plan
+from axe a
+where a.parent is null;
+
+-- Fonction pour supprimer de manière récursive un axe, ses enfants et ses fiches
+create or replace function delete_axe_all(axe_id integer) returns void as
```

Comment:
Le commentaire pourrait être dans `comment is` pour documenter l'API.

Le `or replace` n'est pas nécessaire, autant ne pas le mettre afin que le test de réversibilité soit correct : en effet il passerait même le `drop` ne figurait pas dans le fichier `revert`.